src/lib.cpp

```diff
@@ -34,6 +34,11 @@ struct Orientation {
   float confidence = 0.0f;
 };
 
+struct GetVariableResult {
+  bool success;
+  std::string value;
+};
```

Comment:
I'm OK with merging this as-is, but I think it will make sense to combine this with `OCRResult` in future by making it a templated type (ie. `OCRResult<std::string>` for a fallible operation that returns a string `OCRResult<void>` for a fallible operation that returns nothing).

Reply:
Yes, that approach makes sense to me as well. I was also curious about using standard library facilities like `std::variant`, but I'm not experienced enough in c++ or emscripten to know how to do it right.

Reply:
If https://github.com/emscripten-core/emscripten/issues/8682 is accurate then std::variant might be tricky to use with Emscripten. A struct like the current implementation is nice and simple in comparison, and the performance difference will be negligible.